ts/session/utils/MessageUtils.ts

```diff
@@ -0,0 +1,89 @@
+import { RawMessage } from '../types/RawMessage';
+import { ContentMessage } from '../messages/outgoing';
+import { EncryptionType } from '../types/EncryptionType';
+import * as crypto from 'crypto';
+
+function toRawMessage(device: PubKey, message: ContentMessage): RawMessage {
+  const ttl = message.ttl();
+  const timestamp = message.timestamp;
+  const plainTextBuffer = message.plainTextBuffer();
+
+  // Get EncryptionType depending on message type.
+  // let encryption: EncryptionType;
+
+  // switch (message.constructor.name) {
+  //   case MessageType.Chat:
+  //     encryption = EncryptionType.Signal;
+  //     break;
+  //   case MessageType.SessionReset:
+  //     encryption = EncryptionType
+  // }
+
+  // export enum EncryptionType {
+  //   Signal,
+  //   SessionReset,
+  //   MediumGroup,
+  // }
+
+  // tslint:disable-next-line: no-unnecessary-local-variable
+  const rawMessage: RawMessage = {
+    identifier: message.identifier,
+    plainTextBuffer,
+    timestamp,
+    device: device.key,
+    ttl,
+    encryption: EncryptionType.Signal,
+  };
+
+  return rawMessage;
+}
+
+export enum PubKeyType {
+  Primary = 'priamry',
+  Secondary = 'secondary',
+  Group = 'group',
+}
+
+export class PubKey {
+  private static readonly regex: string = '^0[0-9a-fA-F]{65}$';
+  public readonly key: string;
+  public type?: PubKeyType;
+
+  constructor(pubkeyString: string, type?: PubKeyType) {
+    PubKey.validate(pubkeyString);
+    this.key = pubkeyString;
+    this.type = type;
+  }
+
+  public static from(pubkeyString: string): PubKey {
+    // Returns a new instance if the pubkey is valid
+    if (PubKey.validate(pubkeyString)) {
+      return new PubKey(pubkeyString);
+    }
+
+    throw new Error('Invalid pubkey format');
+  }
+
+  public static validate(pubkeyString: string): boolean {
+    if (pubkeyString.match(PubKey.regex)) {
+      return true;
+    }
+
+    throw new Error('Invalid pubkey format');
```

Comment:
this should return `true/false` and not throw an error. The error should be thrown in the constructor instead.